test/src/tests/generated/paginggroup/models.go

```diff
@@ -157,11 +157,16 @@ func (page *OdataProductResultPage) NextWithContext(ctx context.Context) (err er
 			tracing.EndSpan(ctx, sc, err)
 		}()
 	}
-	next, err := page.fn(ctx, page.opr)
-	if err != nil {
-		return err
+	for {
+		next, err := page.fn(ctx, page.opr)
+		if err != nil {
+			return err
+		}
+		page.opr = next
+		if next.OdataNextLink == nil || (next.Values != nil && len(*next.Values) != 0) {
```

Comment:
I think this also needs to check for an empty next link.  Might be good to generate a helper method "HasNextLink() bool" which would also be used in e.g. `odataProductResultPreparer()`.

Also can't the second condition be rewritten to use `IsEmpty()`?

Reply:
Very reasonable. I will take a look to add the new function `hasNextLink` and I suppose this function could be un-exported.

And the second part of the condition is exactly `!IsEmpty()`